Resources/MessageQueue/MQConnection.py

```diff
@@ -0,0 +1,107 @@
+"""
+Abstract class for management of MQ connections
+"""
+
+import time
+
+from DIRAC import gLogger, S_OK, S_ERROR
+from DIRAC.Core.Utilities.DErrno import EMQUKN
+
+__RCSID__ = "$Id$"
+
+class MQConnectionError( Exception ):
+    def __init__(self,*args,**kwargs):
+        Exception.__init__(self,*args,**kwargs)
+
+class MQConnection( object ):
+  """
+  Abstract class for management of message queue connections
+  Allows to both send and receive messages from a queue
+  """
+
+  MANDATORY_PARAMETERS = [ 'Host', 'Port', 'User', 'VHost', 'MQType' ]
+
+  def __init__( self ):
+    """ Standard constructor
+    """
+
+    self.alive = False
+
+  def on_error( self, headers, message ):
+    """ Default callback function called when an error happens
+    """
+    gLogger.error( message )
+
+  # Rest of functions
+
+  def setParameters( self, parameters ):
+    """ Sets the parameters attribute
+
+    :param dict parameters: dictionary with the parameters for the queue. It should include the following parameters:
+    'Host', 'Port', 'User', 'VH' and 'MQType'. Otherwise, the function will return an error
+    """
+
+    if not parameters:
+      return S_ERROR( EMQUKN, 'Queue parameters are not provided' )
+
+    for param in self.MANDATORY_PARAMETERS:
+      if param not in parameters:
+        return S_ERROR( EMQUKN, "Parameter %s not provided" % param )
```

Comment:
You could make a set difference to return the list of missing parameters ? So in one call we know all the missing params